_spinnaker/automated_rollbacks.md

```diff
@@ -0,0 +1,147 @@
+---
+layout: post
+title: Automated Kubernetes Rollbacks
+order: 140
+---
+{:.no_toc}
+* This is a placeholder for an unordered list that will be replaced with ToC. To exclude a header, add {:.no_toc} after it.
+{:toc}
+
+There are several ways to specify kubernetes deployment rollbacks, and these are based on the complexity and number of artifacts being deployed. We’ll see an example where one artifact is deployed.
+
+### Creating a main deployment pipeline
+
+Let’s start by creating a new pipeline that will deploy a simple nginx container:
+
+![image](/assets/images/aut-rollbacks-deploy-pipeline.png)
+
+We’ll be supplying the manifest directly in the Deploy (Manifest) stage as text for simplicity, but it can also be taken from a version control system like Github:
+
+![image](/assets/images/aut-rollbacks-deploy-manifest-1.png)
+
+This is the sample deployment manifest, which will create two nginx pods in kubernetes under the “german-env1” namespace (the namespace must exist already):
+
+```yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: nginx-deployment
+  namespace: german-env1
+spec:
+  selector:
+    matchLabels:
+      app: nginx
+  replicas: 2
+  template:
+    metadata:
+      labels:
+        app: nginx
+    spec:
+      containers:
+      - name: nginx
+        image: nginx:1.14.0
+        ports:
+        - containerPort: 80
+```
+
+This is all that is required in the deploy pipeline. We can now save and execute manually the pipeline and it should be successful. 
+
+### Generating deployment historic versions for testing rollbacks
+
+We’ll test rollbacks by making a change to the deployment manifest and running again the pipeline, so that we’ll end up with two historic versions of the deployment. First we’ll edit the manifest to change nginx image version from `nginx:1.14.0` to `nginx:1.15.05-alpine` and then run again the pipeline. Now under infrastructure view we’ll see that we have two historic versions of the deployment:
```

Comment:
Recommend changing "running again the pipeline" to "running the pipeline again".  In the second sentence, remove "again" altogether, so it's just "...and then run the pipeline".